### ISSUE TITLE
publishConfig does not overrides after "lerna publish"

### ISSUE THREAD

**Issue by artechventure** [lerna/lerna#3402]
## Current Behavior

publishConfig does not overrides package.json fields after "lerna publish".

original
```
{
  "main": "./src/index.ts",
  "publishConfig": {
    "main": "./dist/index.js"
  }
}
```

uploaded on npm
```
{
  "main": "./src/index.ts",
  "publishConfig": {
    "main": "./dist/index.js"
  }
}
```

## Expected Behavior

package.json should be overwritten and publishConfig should be removed

```
{
  "main": "./dist/index.js",
}

## Steps to Reproduce

<!-- Help us help you by making it easy for us to reproduce your issue! -->

<!-- Can you reproduce this on https://github.com/lerna/repro? -->
<!-- If so, open a PR with your changes and link it below. -->
<!-- If not, please provide a minimal Github repo of your own -->
<!-- At the very least, provide as much detail as possible to help us reproduce the issue -->

<!-- Remove this line -->

1.  add "main" field in publishConfig different with original "main" field.
2.  run "lerna publish"
3.  install uploaded package on npm, and check its package.json

### Failure Logs / Configuration

<!-- Please include any relevant log snippets or files here. -->

<details><summary>lerna.json</summary><p>
<!-- browsers demand the next line be empty -->

```json
<!-- Please paste your `lerna.json` here -->
```

</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```

</p></details>

### Environment

  System:
    OS: macOS 12.1
    CPU: (10) arm64 Apple M1 Max
  Binaries:
    Node: 16.14.0 - /private/var/folders/zr/3tkznhqs6sn26wsynpczk2pxz0g_by/T/xfs-a49a6c75/node
    Yarn: 3.2.1 - /private/var/folders/zr/3tkznhqs6sn26wsynpczk2pxz0g_by/T/xfs-a49a6c75/yarn
  Utilities:
    Git: 2.32.1 - /usr/bin/git

**Comment by artechventure** [comment]
https://github.com/lerna/lerna/blob/e33be4b043f4392311a1b12a1e3de4bdd5e0f42d/utils/pack-directory/lib/pack-directory.js#L55

Yarn, pnpm has its own packing logic, (via yarn pack, pnpm pack).
And inside the pack command, I think there's a logic that overrides package.json's field by publishConfig.
So creating tar like this might not work as expected in yarn, pnpm.

**Comment by sahibjotsaggu** [comment]
@ghiscoding The mentioned issue talks about pnpm. The publishConfig replacement doesn't seem to work with yarn. It works if I do `yarn npm publish` to manually publish a package but not with `lerna publish`

**Comment by ghiscoding** [comment]
@sahibjotsaggu yes it's been supported for a while now in [Lerna-Lite](https://github.com/lerna-lite/lerna-lite) with the help of @artechventure, see [publishConfig overrides](https://github.com/lerna-lite/lerna-lite/tree/main/packages/publish#publishconfig-overrides) and the implementation is not package manager dependent, it works on any of them. 

However Lerna itself does not yet support it, hence why this issue is still open here.

**Comment by elixxx** [comment]
For me this sound like an important feature, to make lerna usable with Typescript right? Can we expect any progress?
